nixos/modules/services/web-apps/peertube/default.nix

```diff
@@ -0,0 +1,177 @@
+{ lib, pkgs, config, ... }:
+
+let
+  name = "peertube";
+  cfg = config.services.peertube;
+
+  uid = config.ids.uids.peertube;
+  gid = config.ids.gids.peertube;
+in
+{
+  options.services.peertube = {
+    enable = lib.mkEnableOption "Enable Peertube’s service";
+
+    user = lib.mkOption {
+      type = lib.types.str;
+      default = name;
+      description = "User account under which Peertube runs";
+    };
+
+    group = lib.mkOption {
+      type = lib.types.str;
+      default = name;
+      description = "Group under which Peertube runs";
+    };
+
+    dataDir = lib.mkOption {
+      type = lib.types.path;
+      default = "/var/lib/${name}";
+      description = ''
+        The directory where Peertube stores its data.
+      '';
+    };
+
+    database = lib.mkOption {
+      type = lib.types.str;
+      default = "peertube_prod";
+      description = ''
+        The Postgres database where Peertube stores its data.
+      '';
+    };
+
+    configFile = lib.mkOption {
+      type = lib.types.path;
+      description = ''
+        The configuration file path for Peertube.
+      '';
+    };
+
+    package = lib.mkOption {
+      type = lib.types.package;
+      default = pkgs.peertube;
+      description = ''
+        Peertube package to use.
+      '';
+    };
+
+    # Output variables
+    systemdStateDirectory = lib.mkOption {
```

Comment:
What is this for? I'm no Nix expert but the assert down seems to make this break if the `dataDir` is set to something outside of `/var/lib`.

Reply:
Can this not be hardcoded to `/var/run/${name}`?